staging/src/k8s.io/kubectl/pkg/cmd/config/create_cluster.go

```diff
@@ -19,14 +19,15 @@ package config
 import (
 	"errors"
 	"fmt"
-	"io"
 	"io/ioutil"
 	"os"
 	"path/filepath"
 
 	"github.com/spf13/cobra"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
 	"k8s.io/client-go/tools/clientcmd"
 	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+	utilcert "k8s.io/client-go/util/cert"
```

Comment:
Nice! I didn't know this existed.

Reply:
hum now I see where you've found this :D

I was going to suggest that we add that CERTIFICATE validation into this CertsFromFile function, but apparently it is used by other parts of the code and would break things, mostly notable:

https://github.com/kubernetes/kubernetes/blob/68e97062de5623210ed48cc91d936ee8511374a4/cmd/kubeadm/app/phases/certs/renewal/readwriter.go#L80 -> which might have PemEncoded Key file

Reply:
We can not call the `certutil.CertsFromFile(certificatePath) `.

```shell
kubectl config set-cluster test-cluster-3 --embed-certs --certificate-authority <(cat /tmp/server.crt)
```
In this case, It read the file twice. Except for the first time, reading again will not get the data.